### DISCUSSION TITLE
"Error: The operation was canceled" when no job was canceled

### DISCUSSION THREAD

**Discussion by sgandon** [community/community#160680] [Actions]
Hello 
We are using using the ARController in our EKS cluster and we are having some issues with our runners deployed using RunnerDeployment. We are using Github cloud (and not the GH server).
Sometimes, and to be honest quite often we are getting our job killed somehow.
The last message in the job logs is the following
```
Error: The operation was canceled.
```
the messages in the job summary is a bit different
```
test / helm_testThe operation was canceled.

The runner has received a shutdown signal. This can happen when the runner service is stopped, or a manually started runner is canceled.
The operation was canceled.
```

And looking at the pods logs it seems that the pod is recieving a SIGTERM message but we don't know why.
```
Runner update in progress, do not shutdown runner.
Downloading 2.302.1 runner
Waiting for current job finish running.
Generate and execute update script.
Runner will exit shortly for update, should be back online within 10 seconds.
Runner update process finished.
Runner listener exit because of updating, re-launch runner after successful update
Update finished successfully.
Restarting runner...

√ Connected to GitHub

Current runner version: '2.302.1'
2023-03-13 10:28:39Z: Listening for Jobs
2023-03-13 10:28:43Z: Running job: build_images
time="2023-03-13T10:28:48.432312191Z" level=info msg="parsed scheme: \"\"" module=grpc
time="2023-03-13T10:28:48.432383100Z" level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
time="2023-03-13T10:28:48.432404757Z" level=info msg="ccResolverWrapper: sending update to cc: {[{localhost  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
time="2023-03-13T10:28:48.432415758Z" level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
2023-03-13 10:29:13.979  NOTICE --- Executing actions-runner-controller's SIGTERM handler.
2023-03-13 10:29:13.980  NOTICE --- Note that if this takes more time than terminationGracePeriodSeconds, the runner will be forcefully terminated by Kubernetes, which may result in the in-progress workflow job, if any, to fail.
2023-03-13 10:29:13.982  NOTICE --- Ensuring dockerd is still running.
CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES
/runner /
2023-03-13 10:29:14.2  NOTICE --- Waiting for the runner to register first.
2023-03-13 10:29:14.30  NOTICE --- Observed that the runner has been registered.

# Runner removal

Failed: Removing runner from the server
Runner "github-actions-runner-large-scwn9-fzjff" is still running a job"
2023-03-13 10:29:16.130  NOTICE --- Waiting for RUNNER_GRACEFUL_STOP_TIMEOUT=15 seconds until the runner agent to stop by itself.
/
2023-03-13 10:29:31.185  NOTICE --- Sending SIGTERM to the actions runner agent (327).
2023-03-13 10:29:31.187  NOTICE --- SIGTERM sent. If the runner is still running a job, you'll probably see "Error: The operation was canceled." in its log.
2023-03-13 10:29:31.192  NOTICE --- Waiting for the actions runner agent to stop.
```

Could anyone help us fix this issue please ?

**Comment by lucetre** [comment]
Are there any updates?

**Comment by maeghan-porter** [comment]
I am having this exact same issue, also running RunnerDeployments in EKS. I was just going to post about this as well. I'm using `summerwind/actions-runner-dind:v2.303.0-ubuntu-22.04` images and SPOT 8vCPU core 32GB memory instances. The instances don't show any significant spike in CPU or memory usage around that time.
My only working theory was that it was because they are SPOT instances and perhaps the instances were getting cycled, however the instance running the container I just had get sigtermed is still around and has been for 5 days, so I'm really at a loss now :(

@sgandon this post was 3 months ago, did you figure out a resolution?

**Comment by MPV** [comment]
Related to this upstream issue?
- https://github.com/actions/runner/issues/2582

**Comment by adefemi171** [comment]
I am also stuck on this as well, it works and then it doesn't on other runs. I have increased the memory and cpu hoping things will change but it just does not work at the moment, any updates?

**Comment by ViniciusBastosTR** [comment]
The Github team should pay more attention to this problem. There are many complaints about this and they are not being resolved.

**Comment by MPV** [comment]
FYI: If it helps, we've seen this issue when nodes **run out of disk** (on both GitHub-hosted and self-hosted runners). The user experience is that the job is "cancelled" and no helpful error about the disk being the cause.

**Comment by ggprod** [comment]
We are seeing this issue as well, could it occur if the associated pod exceeds the cpu or memory limit?

**Comment by hasankfz** [comment]
we are also experiencing this same issue

**Comment by claudiallium** [comment]
i am also experiencing this issue

**Comment by jonwest** [comment]
If you are running on spot instances you will also get this when the nodes are preempted.

**Comment by cb-krishnapatel** [comment]
I've faced a similar issue for a long time. When I deep-dived into this, I got to know that there is a runner registration API called from pod which failed, the failure was due to token expiry. I'm using a GitHub app and after a few hours, this token gets refreshed. is this the reason why?

**Comment by piljaechae** [comment]
Why isn't Github paying attention to this issue? I see a lot of issues related with this, but the only answer that is given is "you're using old version of github runner". This issue is really old but is also occuring on the latest versions.

**Comment by rbellido54** [comment]
In silent pods where code should run,  
SIGTERM cuts, our tasks undone.  
For a year, no answer found,  
We await solutions, hope unbound.

**Reply by VardyNg** [reply]
didn't expect a beautiful poem here

**Comment by braboobssiere** [comment]
got this error yesterday on ubuntu-latest

**Comment by devmonixtechnologies** [comment]
### **The problem happens because of:**

- The runner pod may be automatically restarted by Kubernetes due to updates or deployment changes, which sends SIGTERM to stop the current process.​
- During runner auto-updates or configuration changes, any running jobs are interrupted and canceled by the shutdown signal.
- Resource constraints (like CPU, memory OOM) or controller updates might prematurely stop or restart the runner pods.
- The runner might be unable to properly catch and delay the shutdown during update, especially if docker or other sidecars exit too fast.​

### **We can fix that by doing this steps:**

- Graceful Runner Updates: When using the actions-runner-controller, ensure your update strategy prevents terminating pods with jobs in progress. Use the “grace period” settings in your controller configuration to allow jobs to finish before shutting down the runner.​
- Increase Pod Termination Grace Period: Set higher terminationGracePeriodSeconds for your runner pods to ensure jobs have time to complete before being forcefully killed.
- Disable Auto-Updates During Jobs: Some controller versions allow disabling runner auto-updates while jobs are running; check your Helm chart and ARC configuration for relevant options.​
- Monitor Resources: If OOM or resource limits are causing SIGTERM, increase the memory/CPU requests for your runner pods.​
- Check Controller Versions: Make sure you’re running the latest versions of actions-runner-controller and runner images, as some older versions were overly aggressive in restarting pods during updates.​
- Review Logs: Deep-dive into your pod and Kubernetes controller logs to confirm why SIGTERM is being sent and adjust your deployment to prevent unintentional restarts.

**Comment by olefemmy** [comment]
When I reverted from `ubuntu-latest` to `ubuntu-22.04`, the pipeline seems to be happy.

**Reply by nikitosych** [reply]
gonna give it a shot

**Reply by cyrilchapon** [reply]
@nikitosych confirmed ?

**Reply by nikitosych** [reply]
No. Problem is still not resolved as for me.

**Comment by titanism** [comment]
https://x.com/fwdemail/status/2015782464438866300

**Comment by Feureth** [comment]
In my case, I've just removed the old cache.
Actions -> Cache
